pkg/alibabacloud/eni/types/types.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"github.com/cilium/cilium/pkg/ipam/types"
+)
+
+// Spec is the ENI specification of a node. This specification is considered
+// by the cilium-operator to act as an IPAM operator and makes ENI IPs available
+// via the IPAMSpec section.
+//
+// The ENI specification can either be provided explicitly by the user or the
+// cilium-agent running on the node can be instructed to create the CiliumNode
+// custom resource along with an ENI specification when the node registers
+// itself to the Kubernetes cluster.
+type Spec struct {
+	// InstanceType is the ECS instance type, e.g. "ecs.g6.2xlarge"
+	//
+	// +kubebuilder:validation:Optional
+	InstanceType string `json:"instance-type,omitempty"`
+
+	// AvailabilityZone is the availability zone to use when allocating
+	// ENIs.
+	//
+	// +kubebuilder:validation:Optional
+	AvailabilityZone string `json:"availability-zone,omitempty"`
+
+	// VPCID is the VPC ID to use when allocating ENIs.
+	//
+	// +kubebuilder:validation:Optional
+	VPCID string `json:"vpc-id,omitempty"`
+
+	// CidrBlock is vpc ipv4 CIDR
+	//
+	// +kubebuilder:validation:Optional
+	CidrBlock string `json:"cidr-block,omitempty"`
+
+	// VSwitchs is the ID of vSwitch available for ENI
+	//
+	// +kubebuilder:validation:Optional
+	VSwitchs []string `json:"vswitchs,omitempty"`
```

Comment:
In other code (like instances.go), this seems to be `VSwitches` rather than `VSwitchs`. Can we make these consistent?
```suggestion
	// VSwitchs is the ID of vSwitch available for ENI
	//
	// +kubebuilder:validation:Optional
	VSwitches []string `json:"vswitches,omitempty"`
```

Reply:
sure